components/selection/selection-mixin.js

```diff
@@ -157,7 +157,12 @@ export const SelectionMixin = superclass => class extends RtlMixin(CollectionMix
 		if (e.keyCode < keyCodes.LEFT || e.keyCode > keyCodes.DOWN) return;
 
 		const selectables = Array.from(this._selectionSelectables.values())
-			.filter(item => !item.disabled);
+			.filter(item => !item.disabled)
+			.sort((a, b) => {
+				const idx = a.findIndex((el, idx) => el !== b[idx]);
+				return a[idx].compareDocumentPosition(b[idx]) & Node.DOCUMENT_POSITION_FOLLOWING ? -1 : 1;
+			})
+			.map(p => p.slice(-1)[0]);
```

Comment:
Given two reversed selectable composed paths, `[a,d,e]` and `[a,b,c]`, compare the DOM order of the first selectable ancestors that do not match, so `d.compareDocumentPosition(b)` results in the properly sorted selectables `[c, e]`.